tutorials/sphinx_tuto/tensorclass_imagenet.py

```diff
@@ -117,30 +204,40 @@ class ImageNetData:
     targets: torch.Tensor
 
     @classmethod
-    def from_dataset(cls, dataset, device=None):
+    def from_dataset(cls, dataset):
         data = cls(
             images=MemmapTensor(
-                len(dataset), *dataset[0][0].squeeze().shape, dtype=torch.float32
+                len(dataset),
+                *dataset[0][0].squeeze().shape,
+                dtype=torch.uint8,
             ),
             targets=MemmapTensor(len(dataset), dtype=torch.int64),
             batch_size=[len(dataset)],
-            device=device,
         )
-        for i, (image, target) in enumerate(dataset):
-            data[i] = cls(images=image, targets=torch.tensor(target), batch_size=[])
-        return data
+        data = data.memmap_()
+
+        batch = 32
+        dl = DataLoader(dataset, batch_size=batch, num_workers=NUM_WORKERS)
+        i = 0
+        pbar = tqdm.tqdm(total=len(dataset))
+        for image, target in dl:
+            _batch = image.shape[0]
+            pbar.update(_batch)
+            data[i : i + _batch] = cls(
+                images=image, targets=target, batch_size=[_batch]
+            )
+            i += _batch
```

Comment:
Does the `tqdm` stuff here make this too confusing?

Reply:
Do you mean: do we want it there?
The thing is that otherwise you can wait for some minutes without a progress bar which is a bit mysterious for the users...

Reply:
Yeah that's fair. Let's leave it then. I guess most people are probably pretty familiar with it anyway.